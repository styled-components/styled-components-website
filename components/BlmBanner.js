import React from 'react';

import styled from 'styled-components';
import rem from '../utils/rem';

const Banner = styled.a`
  background: black;
  color: white;
  display: block;
  text-align: center;
  font-size: ${rem(15)};
  font-weight: 500;
  justify-content: center;
  padding: 0.75rem 0.5rem;
  width: 100%;
`;

export const BlmBanner = () => {
  return (
    <Banner href="https://support.eji.org/give/153413/#!/donation/checkout">
      #BlackLivesMatter ✊🏿 <span style={{ textDecoration: `underline` }}>Support the Equal Justice Initiative</span>
    </Banner>
  );
};
