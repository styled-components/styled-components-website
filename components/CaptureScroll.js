import React from 'react'; // eslint-disable-line
import { findDOMNode } from 'react-dom';

let isMobile;
let lastWheelTimestamp;

if (typeof window !== 'undefined' && window.matchMedia) {
  isMobile = window.matchMedia(`(max-width: ${1000 / 16}em)`).matches;

  if (!isMobile) {
    window.addEventListener('wheel', ({ timeStamp }) => {
      lastWheelTimestamp = timeStamp;
    });
  }
}

const captureScroll = (Component) => {
  if (isMobile) {
    return Component;
  }

  class CaptureScroll extends React.Component {
    onScroll = (evt) => {
      // Don't access window wheel listener
      evt.stopImmediatePropagation();

      const { timeStamp, deltaY } = evt;
      const { offsetHeight, scrollHeight, scrollTop } = this.node;

      // If the window is being scrolled, don't scroll the captured scroll area
      if (timeStamp - lastWheelTimestamp <= 400) {
        lastWheelTimestamp = timeStamp;

        evt.preventDefault();
        window.scrollBy(0, deltaY);
        return;
      }

      const maxScrollTop = scrollHeight - offsetHeight;

      // Has the scroll area reached it's beginning/end
      const hasReachedTop = deltaY < 0 && scrollTop === 0;
      const hasReachedBottom = deltaY > 0 && scrollTop >= maxScrollTop;

      // Is the trajectory overshooting the scroll area
      const isReachingTop = scrollTop + deltaY <= 0;
      const isReachingBottom = scrollTop + deltaY >= maxScrollTop;

      if (hasReachedTop || hasReachedBottom || isReachingTop || isReachingBottom) {
        evt.preventDefault();
      }

      // If we're overshooting, we need to set the maximum available position
      if (isReachingTop || isReachingBottom) {
        this.node.scrollTop = isReachingTop ? 0 : maxScrollTop;
      }
    };

    onResize = () => {
      isMobile = window.matchMedia(`(max-width: ${1000 / 16}em)`).matches;
      if (isMobile) {
        this.node.removeEventListener('wheel', this.onScroll);
      } else {
        this.node.addEventListener('wheel', this.onScroll);
      }
    };

    componentDidMount() {
      // eslint-disable-next-line react/no-find-dom-node
      this.node = findDOMNode(this.ref);

      this.node.addEventListener('wheel', this.onScroll);
      window.addEventListener('resize', this.onResize);
    }

    componentWillUnmount() {
      this.node.removeEventListener('wheel', this.onScroll);
      window.removeEventListener('resize', this.onResize);
    }

    render() {
      return <Component {...this.props} ref={(x) => (this.ref = x)} />;
    }
  }

  return CaptureScroll;
};

export default captureScroll;
