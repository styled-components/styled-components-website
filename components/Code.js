import styled from 'styled-components'
import rem from '../utils/rem'
import { lightGrey } from '../utils/colors'
import { monospace } from '../utils/fonts'

import '../utils/prismTemplateString'
import { Editor } from 'react-live'

const Code = styled.span`
  font-family: ${monospace};
  font-weight: 500;
`

export default Code
