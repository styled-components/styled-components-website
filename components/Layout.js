import styled, { css } from 'styled-components';
import rem from '../utils/rem';
import { mobile } from '../utils/media';
import { sidebarWidth } from '../utils/sizes';
import { bodyFont, headerFont } from '../utils/fonts';
import { blmMetal } from '../utils/colors';

export const Container = styled.div`
  padding-left: ${rem(sidebarWidth)};

  ${mobile(css`
    padding-left: 0;
  `)};
`;

export const Content = styled.div`
  width: ${rem(1024)};
  max-width: 100%;
  margin: 0 auto;
  padding: ${rem(90)} ${rem(40)} 0 ${rem(40)};
  box-sizing: border-box;
  font-family: ${bodyFont};
  transition: transform 150ms ease-out;

  ${mobile(css`
    padding: ${rem(100)} ${rem(20)} ${rem(30)} ${rem(20)};
    transform: translateX(${(p) => (p.moveRight ? rem(sidebarWidth) : 0)});
  `)};

  ${(p) =>
    p.hero &&
    css`
      font-family: ${headerFont};
      width: 75rem;
    `};
`;

export const Title = styled.h1`
  display: block;
  text-align: left;
  width: 100%;
  color: ${blmMetal};
  font-size: ${rem(42)};
  font-weight: bold;
  font-family: ${headerFont};
`;

export const Header = styled.h2`
  font-size: ${rem(32)};
  font-weight: 600;
  font-family: ${headerFont};
  margin: 2em 0 0.75em;
`;

export const SubHeader = styled.h3`
  display: block;
  margin: 2em 0 0.75em;
  font-size: ${rem(24)};
  font-weight: 600;
  font-family: ${headerFont};
`;

export const TertiaryHeader = styled.h4`
  display: block;
  margin: 2em 0 0.75em;
  font-size: ${rem(18)};
  font-weight: 600;
  font-family: ${headerFont};
`;
