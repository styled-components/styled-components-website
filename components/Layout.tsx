import styled, { css } from 'styled-components';
import { blmMetal } from '../utils/colors';
import { bodyFont, headerFont } from '../utils/fonts';
import { mobile } from '../utils/media';
import rem from '../utils/rem';
import { sidebarWidth } from '../utils/sizes';

export const Container = styled.div`
  padding-left: ${rem(sidebarWidth)};

  ${mobile(css`
    padding-left: 0;
  `)};
`;

export const Content = styled.div<{ $hero?: boolean; $moveRight?: boolean }>`
  box-sizing: border-box;
  font-family: ${bodyFont};
  margin: 0 auto;
  max-width: 120ch;
  padding: ${rem(90)} ${rem(60)} 0 ${rem(60)};
  transition: transform 150ms ease-out;

  @layer base {
    p,
    li {
      max-width: 100ch;
    }
  }

  ${p =>
    mobile(css`
      padding: ${rem(100)} ${rem(36)} ${rem(30)} ${rem(36)};
      transform: translateX(${p.$moveRight ? rem(sidebarWidth) : 0});
    `)};

  ${p =>
    p.$hero &&
    css`
      font-family: ${headerFont};
    `};
`;

export const Title = styled.h1`
  text-align: left;
  width: 100%;
  color: ${blmMetal};
  font-size: ${rem(42)};
  font-weight: bold;
  font-family: ${headerFont};

  + h2 {
    margin-top: -0.5em;
  }
`;

export const Header = styled.h2`
  font-size: ${rem(32)};
  font-weight: 600;
  font-family: ${headerFont};
  margin: 2em 0 1em;

  + h3 {
    margin-top: -0.5em;
  }
`;

export const SubHeader = styled.h3`
  margin: 2em 0 1em;
  font-size: ${rem(24)};
  font-weight: 600;
  font-family: ${headerFont};

  + h4 {
    margin-top: -0.5em;
  }
`;

export const TertiaryHeader = styled.h4`
  margin: 2em 0 1em;
  font-size: ${rem(18)};
  font-weight: 600;
  font-family: ${headerFont};
`;
