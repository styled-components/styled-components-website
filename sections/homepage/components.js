import styled, { css } from 'styled-components';

export const AlignCenter = styled.div`
  text-align: center;
`;

export const Badge = styled.img`
  margin: 0 0.5em 1em;
  height: 1.5em;
`;

export const ExampleButton = styled.button`
  border-radius: 3px;
  padding: 0.25em 1em;
  margin: 0.5em 1em;
  background: transparent;
  color: palevioletred;
  border: 2px solid palevioletred;

  ${p =>
    p.primary &&
    css`
      background: palevioletred;
      color: white;
    `};
`;

export const SecondButton = styled.button`
  border-radius: 3px;
  padding: 0.25em 1em;
  background: transparent;
  color: palevioletred;
  border: 2px solid palevioletred;
`;
