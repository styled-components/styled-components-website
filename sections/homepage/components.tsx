import styled, { css } from 'styled-components';
import { palepink } from '../../utils/colors';

export const AlignCenter = styled.div`
  text-align: center;
`;

export const Badge = styled.img`
  margin: 0 0.5em 1em;
  height: 1.5em;
`;

export const ExampleButton = styled.button<{ $primary?: boolean }>`
  border-radius: 3px;
  padding: 0.25em 1em;
  margin: 0.5em 1em;
  background: transparent;
  color: ${palepink};
  border: 2px solid ${palepink};

  ${p =>
    p.$primary &&
    css`
      background: ${palepink};
      color: white;
    `};
`;

export const SecondButton = styled.button`
  border-radius: 3px;
  padding: 0.25em 1em;
  background: transparent;
  color: ${palepink};
  border: 2px solid ${palepink};
`;
