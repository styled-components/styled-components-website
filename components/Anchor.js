import styled, { css } from 'styled-components'
import rem from '../utils/rem'

import LinkIcon from 'react-octicons-svg/dist/LinkIcon'
import { Header, SubHeader, TertiaryHeader } from './Layout'
import { mobile } from '../utils/media'

const InvisibleAnchor = styled.div.attrs({
  'aria-hidden': true
})`
  position: relative;
  display: block;
  visibility: hidden;
  height: 0;

  top: ${rem(-70)};

  ${mobile(css`
    top: ${rem(-90)};
  `)}
`

const Anchor = styled.a`
  display: none;
  position: absolute;
  left: 0;
  color: inherit;
`

const AnchorIcon = styled(LinkIcon).attrs({
  width: null,
  height: null
})`
  width: ${rem(20)};
  opacity: 0.7;
  margin-top: ${rem(-5)};

  &:hover {
    opacity: 0.9;
  }
`

const AnchorHeader = styled(Header)`
  position: relative;
  margin-left: ${rem(-30)};
  padding-left: ${rem(30)};

  ${mobile(css`
    margin-left: 0;

    /* stylelint-disable-next-line */
    ${Anchor} {
      display: inline-block;
    }
  `)}

  &:hover ${Anchor} {
    display: inline-block;
  }
`

const AnchorSubHeader = AnchorHeader.withComponent(SubHeader)
const AnchorTertiaryHeader = AnchorHeader.withComponent(TertiaryHeader)

const Link = ({ children, level, id }) => {
  let Child = AnchorHeader

  switch(level) {
    case 3:
      Child = AnchorSubHeader
      break
    case 4:
      Child = AnchorTertiaryHeader
      break
    default:
      break
  }

  return (
    <Child>
      <InvisibleAnchor id={id} />

      <Anchor href={`#${id}`}>
        <AnchorIcon />
      </Anchor>

      {children}
    </Child>
  )
}

export default Link
