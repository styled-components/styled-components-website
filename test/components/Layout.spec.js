import React from 'react';
import renderer from 'react-test-renderer';
import { Container, Content, Title, Header, SubHeader } from '../../components/Layout';

test('Container renders correctly', () => {
  const tree = renderer.create(<Container />).toJSON();

  expect(tree).toMatchSnapshot();
});

test('Content renders correctly', () => {
  const tree = renderer.create(<Content />).toJSON();

  expect(tree).toMatchSnapshot();
});

test('Title renders correctly', () => {
  const tree = renderer.create(<Title />).toJSON();

  expect(tree).toMatchSnapshot();
});

test('Header renders correctly', () => {
  const tree = renderer.create(<Header />).toJSON();

  expect(tree).toMatchSnapshot();
});

test('SubHeader renders correctly', () => {
  const tree = renderer.create(<SubHeader />).toJSON();

  expect(tree).toMatchSnapshot();
});
