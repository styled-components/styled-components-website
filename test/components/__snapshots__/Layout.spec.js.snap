// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders correctly 1`] = `
.c0 {
  padding-left: 16.666666666666668rem;
}

@media (max-width:62.5em) {
  .c0 {
    padding-left: 0;
  }
}

<div
  className="c0"
/>
`;

exports[`Content renders correctly 1`] = `
.c0 {
  width: 56.888888888888886rem;
  max-width: 100%;
  margin: 0 auto;
  padding: 5rem 2.2222222222222223rem 1.6666666666666667rem 2.2222222222222223rem;
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  -webkit-transition: -webkit-transform 150ms ease-out;
  -webkit-transition: transform 150ms ease-out;
  transition: transform 150ms ease-out;
}

@media (max-width:62.5em) {
  .c0 {
    padding: 3.888888888888889rem 1.1111111111111112rem 1.6666666666666667rem 1.1111111111111112rem;
    -webkit-transform: translateX(0);
    -ms-transform: translateX(0);
    transform: translateX(0);
  }
}

<div
  className="c0"
/>
`;

exports[`Header renders correctly 1`] = `
.c0 {
  font-size: 1.7777777777777777rem;
  font-weight: 500;
  font-family: "Avenir Next",-apple-system,BlinkMacSystemFont,"Segoe UI",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
}

<h2
  className="c0"
/>
`;

exports[`SubHeader renders correctly 1`] = `
.c0 {
  display: block;
  margin: 1.9444444444444444rem 0 1.2222222222222223rem 0;
  font-size: 1.3333333333333333rem;
  font-weight: 500;
  font-family: "Avenir Next",-apple-system,BlinkMacSystemFont,"Segoe UI",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
}

<h3
  className="c0"
/>
`;

exports[`Title renders correctly 1`] = `
.c0 {
  display: block;
  text-align: left;
  width: 100%;
  color: rgb(243,182,97);
  font-size: 2.3333333333333335rem;
  font-weight: bold;
  font-family: "Avenir Next",-apple-system,BlinkMacSystemFont,"Segoe UI",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
}

<h1
  className="c0"
/>
`;
